src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -176,139 +180,206 @@ public int updateInternal(Position p) {
-
-    String insertSql;
-    if (DaoUtils.isMsSql()) {
-      insertSql = "/* positionRemovePerson.insert1 */ INSERT INTO \"peoplePositions\" "
-          + "(\"positionUuid\", \"personUuid\", \"createdAt\") VALUES(NULL, "
-          + "(SELECT TOP(1) \"personUuid\" FROM \"peoplePositions\""
-          + " WHERE \"positionUuid\" = :positionUuid ORDER BY \"createdAt\" DESC),"
-          + " :createdAt)";
-    } else {
-      insertSql = "/* positionRemovePerson.insert1 */ INSERT INTO \"peoplePositions\" "
-          + "(\"positionUuid\", \"personUuid\", \"createdAt\") VALUES(NULL, "
-          + "(SELECT \"personUuid\" FROM \"peoplePositions\""
-          + " WHERE \"positionUuid\" = :positionUuid ORDER BY \"createdAt\" DESC LIMIT 1),"
-          + " :createdAt)";
-    }
-    getDbHandle().createUpdate(insertSql).bind("positionUuid", positionUuid)
-        .bind("createdAt", DaoUtils.asLocalDateTime(now)).execute();
+  @InTransaction
+  public int insertPeoplePositionWithoutPerson(String positionUuid) {
+    final Instant now = Instant.now();
+    return insertPeoplePositionWithoutPerson(positionUuid, now);
+  }
 
-    final int nr = getDbHandle()
-        .createUpdate("/* positionRemovePerson.insert2 */ INSERT INTO \"peoplePositions\" "
+  private int insertPeoplePositionWithoutPerson(String positionUuid, Instant now) {
+    return getDbHandle()
+        .createUpdate("/* personEmptyPosition.insert */ INSERT INTO \"peoplePositions\" "
             + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
             + "VALUES (:positionUuid, NULL, :createdAt)")
         .bind("positionUuid", positionUuid)
         // Need to ensure this timestamp is greater than previous INSERT.
         .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
+  }
+
+  @InTransaction
+  public int insertPeoplePositionWithoutPosition(String personUuid) {
+    final Instant now = Instant.now();
+    return insertPeoplePositionWithoutPosition(personUuid, now);
+  }
+
+  private int insertPeoplePositionWithoutPosition(String personUuid, Instant now) {
+
+    return getDbHandle()
+        .createUpdate("/* positionEmptyPerson.insert */ INSERT INTO \"peoplePositions\" "
+            + "(\"positionUuid\", \"personUuid\", \"createdAt\") "
+            + "VALUES (NULL, :personUuid, :createdAt)")
+        .bind("personUuid", personUuid)
+        // Need to ensure this timestamp is greater than previous INSERT.
+        .bind("createdAt", DaoUtils.asLocalDateTime(now.plusMillis(1))).execute();
+  }
+
+  @InTransaction
+  public int removePersonFromPosition(String positionUuid) {
+    final Instant now = Instant.now();
+    return removePersonFromPosition(positionUuid, now);
+  }
+
+  private int removePersonFromPosition(String positionUuid, Instant now) {
+    int numRows = 0;
+    numRows += emptyPosition(positionUuid, now);
+
+    PersonPositionHistory currentPositionPerson = getDbHandle()
```

Comment:
This declaration (and many others) could (and should) be `final`.

Reply:
Only where necessary was numRows used and defined as final.

Reply:
As @gjvoosten notes, this should be changed to
```
final PersonPositionHistory currentPositionPerson = getDbHandle().....
```

Reply:
I've added the keyword